src/py42/services/fileevent.py

```diff
@@ -24,14 +30,34 @@ def search(self, query):
         Returns:
             :class:`py42.response.Py42Response`: A response containing the query results.
         """
+        # deprecation warning for v1 file events
+        if not settings.use_v2_file_event_data:
+            warn(
+                "V1 file events are deprecated.  Set 'py42.settings.use_v2_file_event_data = True' to use V2 file events.",
+                DeprecationWarning,
+                stacklevel=2,
+            )
+
+        # check if query type matches settings
+        if not isinstance(query, str):
+            if query.sort_key == "eventId" and settings.use_v2_file_event_data:
+                raise Py42Error(
+                    "Bad Request. You cannot use a V1 query with V2 settings enabled. For more details on working with V2 file event data, see the 'V2 File Events' user guide at https://py42docs.code42.com/."
```

Comment:
did we want to link to the guide directly here?

Reply:
I wasn't sure if links would make sense in something like an error message? If we think that would be appropriate we certainly could though

Reply:
Yeah, I'm of the mindset that error messages should be as helpful as possible, so the link is good, I was just saying we should link directly to the guide: https://py42docs.code42.com/en/stable/userguides/v2apis.html like we do in the docstrings.